lisa/util/parallel.py

```diff
@@ -2,29 +2,66 @@
 # Licensed under the MIT license.
 
 from concurrent.futures import FIRST_COMPLETED, Future, ThreadPoolExecutor, wait
-from typing import Any, Callable, Generic, List, Optional, TypeVar
+from typing import Any, Callable, Dict, Generic, List, Optional, TypeVar
+
+from lisa.util.logger import Logger, get_logger
+from lisa.util.perf_timer import create_timer
 
 from . import LisaException
 
 T_RESULT = TypeVar("T_RESULT")
 
 
+class Task(Generic[T_RESULT]):
+    def __init__(
+        self,
+        task_id: int,
+        task: Callable[[], T_RESULT],
+        parent_logger: Optional[Logger],
+    ) -> None:
+        self.id = task_id
+        self._task = task
+        self._log = get_logger("Task", str(self.id), parent_logger)
+        self._log.debug(f"Generate task: {self}")
```

Comment:
How about move this line into `__call__`? It's more accurate to see when the task start to run actually.

Reply:
```python
def __call__(self) -> T_RESULT:
    self._timer = create_timer()
    # log start...
    output = self._task()
    # long end...
    return output
```